GitHub Action CodeQuality v1

dmetzner opened the pull request:
Code Quality Checks v1
    
    only 1 CodeQuality GitHub Action:
     - js code style checks
     - php statistics, code style, static checks
    
    - add phpstan static checks
    - add rector for automated refactoring
    
    - Controllers refactored with rector + added datatypes by hand

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing, if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
`TODO: add additional information about testruns here`

dmetzner committed:
Code Quality Checks v1

only 1 CodeStyle GitHub Action:
 - js code style checks
 - php statistics, code style, static checks

- add phpstan static checks
- add rector for automated refactoring

- Controllers refactored with rector + added datatypes by hand

HCrane commented on src/Catrobat/Controller/Web/ProfileController.php:
```diff
@@ -25,34 +24,28 @@
 {
   const MIN_PASSWORD_LENGTH = 6;
+
   const MAX_PASSWORD_LENGTH = 32;
```
Whaat? 🎉

HCrane commented on src/Catrobat/Controller/Web/ProgramController.php:
```diff
@@ -102,29 +89,22 @@ public function programRemixGraphAction(Request $request, $id, RemixManager $rem
-   * @Route("/program/{id}", name="program_depricated")
-   * @Route("/details/{id}", name="catrobat_web_detail", methods={"GET"})
+   * @Route("/project/{id}", name="program")
+   * @Route("/project/{id}", name="program")
```
Revert please because this would break legacy links.

dmetzner committed:
Code Quality v2 - SPEC to UNIT

- Migrate and refactored PhpSpec tests to PhpUnit tests
- Remove PhpSpec Packages
- Only 1 Test Workflow (GA)

- fix codebase to pass phpstan lvl1

- fix tests namespace (needed for phpstan)

- Add tests folder to fixer & phpstan
- refactore (unit) tests to pass phpstan lvl1

- add more Php7.4 static types

- Dockerfile install from dist to improve performance

HCrane reviewed (approved):
💯
